crates/webidl/src/lib.rs

```diff
@@ -668,10 +669,25 @@ impl<'src> FirstPassRecord<'src> {
             let mut doc = doc.clone();
             self.append_required_features_doc(&method, &mut doc, &[]);
             method.doc_comment = doc;
+            self.add_deprecated(data, &mut method.function.rust_attrs);
             program.imports.push(wrap_import_function(method));
         }
     }
 
+    fn add_deprecated(&self, data: &InterfaceData<'src>, dst: &mut Vec<syn::Attribute>) {
+        if !data.deprecated {
+            return
+        }
+        let note = format!("This API is deprecated as it was found to not \
+                            work in all browsers, typically because the bound \
+                            class doesn't exist in all browsers. You can \
+                            probably stop using this class and instead \
+                            use the class that inherits from this, which \
+                            should include all the same methods and work \
+                            in all browsers in theory!");
```

Comment:
It would be nice if we could tailor the deprecation message for the particular interfaces. Something like:

```webidl
// BaseAudioContext.webidl

[Deprecated="reason why this is deprecated detailed here..."]
interface BaseAudioContext {
  // ...
};
```

Reply:
Sure yeah, I can upate this